### DISCUSSION TITLE
Layout placeholders

### DISCUSSION THREAD

**Discussion by antonk52** [facebook/docusaurus#6179] [Ideas]
Currently there is no way to customise existing components (DocItem is a good example). To be more specific inserting custom components within major layout components. The only way can customise these components is to overwrite a given component in your projects/preset while taking a responsibility to manually keep it in sync with the component from docusaurus itself. This process is suboptimal and it would be nice to have a few placeholder(`() => null`) components that can be overwritten by customers who could "insert" their project specific stuff without having to essentially fork docusaurus' component. A couple of good candidates can be Pre/Post Content for DocItem, that can be utilised for comments section, polls and other project specific items.

**Comment by Josh-Cena** [comment]
Hey, here might be a useful thread for you: https://github.com/facebook/docusaurus/issues/5251

Basically, if you just want placeholder components inserted before/after some block, you can usually do so through [wrapping components](https://docusaurus.io/docs/next/using-themes#wrapping-theme-components). If that doesn't work out (e.g. if the place you are inserting your component in is in the middle of a bigger component), you can report your use-case in #6116 and we will see how we can refactor our theme to cope with your needs.

**Reply by antonk52** [reply]
Thank you for the response. Wrapping components can be a good solution for simple cases.

In my case we have to add additional things in `DocItem` file between DocVersionBadge, DocContent and DocItemFooter. In other words a wrapper component for `DocContent` would be perfect in our scenario since we also try to insert an item in between external title and `DocContent` when title is external.

**Reply by slorber** [reply]
@antonk52 we can definitively extract a DocContent theme item, that makes perfect sense to enable you to do so.

Please report your customization use-case here, it's important for us to understand what is your goal exactly: https://github.com/facebook/docusaurus/discussions/5468

**Reply by slorber** [reply]
@antonk52 when I asked to report the customization usecase, I expected something like "I want to add a comment system to my articles", not "we can't wrap X comp" :) What is the final outcome you want, not what is the technical detail that you think prevents you from achieving it.

**Reply by Josh-Cena** [reply]
https://xyproblem.info/ 😝

**Reply by antonk52** [reply]
my bad, let me edit the comment 😅

**Reply by slorber** [reply]
thanks ;)
